lib/Sema/TypeCheckType.cpp

```diff
@@ -2722,6 +2726,15 @@ SILParameterInfo TypeResolver::resolveSILParameter(
     checkFor(TypeAttrKind::TAK_guaranteed,
              ParameterConvention::Direct_Guaranteed);
 
+    // SWIFT_ENABLE_TENSORFLOW
+    if (attrs.has(TAK_nondiff)) {
+      // TODO: We could diagnose @nondiff on a non-@autodiff function, but we'd
```

Comment:
I think this TODO is actually unnecessary, because it's already being diagnosed here: https://github.com/marcrasi/swift/blob/f9cddc12a86cb246eb6a703d9ce8724534d2db6e/lib/Sema/TypeCheckType.cpp#L2374